File: scripts/munge.py
#!/usr/bin/python
from math import *
from sys import argv
import re
import sys
from sys import stderr
def fdiv(a,b):
    return float(a)/float(b)

def av(l):
    s = sum(l)
    n = len(l)
    av = fdiv(s,n);
    dev = 0;
    for i in l:
        dev += (av - i)**2;
    dev = sqrt(dev/n);
    return (av,dev);
def join(s,l):
    return s.join(map(lambda x:str(x), l))

vals = {}
def put(f,p,t):
    if not p in vals:
        vals[p] = {}
    fs = vals[p]
    if not f in fs:
        fs[f] = []
    fs[f].append(t)

DEV=False
if '-std' in argv:
    DEV=True
    argv.remove('-std')

title = "No. pixel, "
for f in argv[1:]:
    fh = open(f);
    fh.readline()
    for l in fh.readlines():
        #(func,file,x,y,pixels,rt,ut,st,mask,total) = re.split(',\s*/',l)
        l = re.split(',\s*',l);
        #print l
        
        ut = l[6]; st = l[7]; pixels = l[4]
        total = float(ut) + float(st);
        put(f,pixels,total)
    if DEV:
        title += "%s Average Time, %s StdDev, " %(f,f)
    else: 
        title += "%s Average Time, " %(f)
        
print title
for pixels in vals.keys():
    res = []
    for f in sys.argv[1:]:
        #for f in vals[pixels].keys():
        if f in vals[pixels]:
           #stderr.write("%s %s pixels %i samples\n"%(f[:5],pixels,len(vals[pixels][f])))
           times = vals[pixels][f];
           (a,dev) = av(times)
           #print pixels,f,times;
        else:
	    a,dev = "NA","NA" 
        stderr.write("%s av:%f dev:%f\n"%(f,a,dev));
        res.append(a)
        if DEV:
            res.append(dev)
    print str(pixels)+','+ join(",",res)

File: scripts/munge.pl
#!/usr/bin/perl -w
sub av {
    $r = shift;
    $n = $sum = 0;
    for $i (@{$r}) {
        $sum += $i;
        $n++;
    }
    $av = $sum / $n;
    $dev = 0;
    for $i (@{$r}) {
        $dev += ($i - $av)**2;
    }
    $dev = sqrt($dev);
    return ($sum/$n,$dev);
}

for $f (@ARGV) {
    open( F, $f) or die "Cannot open $f $!";
    print $f;
    <F>; # skip header
    while (<F>) {
        ($func,$file,$x,$y,$pixels,$rt,$ut,$st,$mask,$total) = split /,\s*/;
        $total = $ut + $st;
        $r = $vals{$pixels};
        %res = ($r) ? %{$r} : (); 
        push @{$res{$f}}, $total;
        $vals{$pixels} = \%res;
    }
}

print "# No. pixels, Average Time, Standard Deviationn\n";
foreach $p (keys %vals) {
    %res = %{$vals{$p}};
    foreach $f (keys %res) {
        #print "$p -> $f\n";
        @times = @{$res{$f}};
        
        ($m,$dev) = av(\@times);
        print "$p -> $f -> |@times|\n";
    }


    #print "$p -> @{$vals{$p}} -> $m, $dev\n";
    #print "$p, $m, $dev\n";
    #print "$p $m $dev\n";
    #}
}

